sstables/sstables.cc

```diff
@@ -911,13 +911,24 @@ file_writer::~file_writer() {
 
 void file_writer::close() {
     assert(!_closed && "file_writer already closed");
+    std::exception_ptr ex;
+    try {
+        _out.flush().get();
```

Comment:
We can rename the function to flush_and_close().

btw

Reply:
yes btw. If called from the destructor during exception handling, we shouldn't flush.

The destructor path should look something like

if (during rollback) {
    close without flushing, ignore errors
} else {
    warn if not closed
}

Reply:
> If called from the destructor during exception handling, we shouldn't flush.

@avikivity , but it doesn't:

```c++
file_writer::~file_writer() {
    if (_closed) {
        return;
    }
    try {       
        // close() should be called by the owner of the file_writer.
        // However it may not be called on exception handling paths
        // so auto-close the output_stream so it won't be destructed while open.
        _out.close().get();
    } catch (...) {
        sstlog.warn("Error while auto-closing {}: {}. Ignored.", get_filename(), std::current_exception());
    }
}       
```

it calls `_out.close()` not `this->close()`

Reply:
> Doesn't close() imply flush?

Not necessarily. The file_data_sink_impl doesn't flush() underlying fd on its close(). Similarly e.g. posix_file_impl's close() just closes the descriptor, while flush() calls fdatasync() (@tgrabiec )